src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/RuleManagerImpl.java

```diff
@@ -0,0 +1,146 @@
+            throw new IllegalArgumentException("Should use only one switch.");
+        }
+        if (adapter.getSwitchProperties().size() != 1) {
+            throw new IllegalArgumentException("Should use only one switch properties.");
+        }
+    }
+
+    private List<Command> buildDefaultRules(Switch sw, SwitchProperties switchProperties) {
+        return getDefaultRuleGenerators(switchProperties).stream()
+                .flatMap(g -> g.generateCommands(sw).stream())
+                .collect(toList());
+    }
+
+    private List<RuleGenerator> getDefaultRuleGenerators(SwitchProperties switchProperties) {
+        List<RuleGenerator> generators = new ArrayList<>();
+        generators.add(defaultRulesFactory.getDropRuleGenerator(DROP_RULE_COOKIE, INPUT_TABLE_ID));
+        generators.add(defaultRulesFactory.getVerificationRuleGenerator(true));
+        generators.add(defaultRulesFactory.getVerificationRuleGenerator(false));
+        // TODO: add other rules
+
+        if (switchProperties.isMultiTable()) {
+            generators.add(defaultRulesFactory.getDropRuleGenerator(
+                    MULTITABLE_INGRESS_DROP_COOKIE, INGRESS_TABLE_ID));
+            generators.add(defaultRulesFactory.getDropRuleGenerator(
+                    MULTITABLE_TRANSIT_DROP_COOKIE, TRANSIT_TABLE_ID));
+            generators.add(defaultRulesFactory.getDropRuleGenerator(
+                    MULTITABLE_POST_INGRESS_DROP_COOKIE, POST_INGRESS_TABLE_ID));
+
+            // TODO: add other rules
+        }
+
+        return generators;
+    }
+
+    private List<Command> buildFLowRules(Map<SwitchId, Switch> switches,
+                                         Map<SwitchId, SwitchProperties> switchProperties,
+                                         Collection<FlowPath> flowPaths) {
+        return flowPaths.stream()
+                .flatMap(flowPath -> buildFLowRules(switches, switchProperties, flowPath).stream())
+                .collect(Collectors.toList());
+    }
+
+    private List<Command> buildFLowRules(Map<SwitchId, Switch> switches,
+                                         Map<SwitchId, SwitchProperties> switchProperties,
+                                         FlowPath flowPath) {
+        List<Command> result = new ArrayList<>();
+
+        SwitchId srcSwitchId = flowPath.getSrcSwitchId();
+        if (switches.containsKey(srcSwitchId)) {
+            result.addAll(buildIngressCommands(switches.get(srcSwitchId), switchProperties.get(srcSwitchId), flowPath));
+        }
+        // todo: add transit, egress and other flow rules
+
+        return result;
+    }
+
+    private List<Command> buildIngressCommands(Switch sw, SwitchProperties switchProperties, FlowPath flowPath) {
+        List<RuleGenerator> generators = new ArrayList<>();
+
+        generators.add(flowRulesFactory.getIngressRuleGenerator(flowPath));
+        // todo: add arp, lldp, flow loop, flow mirror, etc
+
+        return generators.stream()
```

Comment:
Initial plan was to resolve all generators inside the lib. It will simplify local run and use of the lib from any calling component. I don't see scenarios where caller should affect in any way not-resolved generators.